clang/test/SemaSYCL/sycl-cconv.cpp

```diff
@@ -1,7 +1,11 @@
 // RUN: %clang_cc1 -triple spir64-unknown-windows-sycldevice -aux-triple x86_64-pc-windows-msvc -fsycl-is-device -fsyntax-only -verify %s
 
-//expected-warning@+1 {{'__cdecl' calling convention is not supported for this target}}
+// expected-no-warning@+1
 __inline __cdecl int printf(char const* const _Format, ...) { return 0; }
+// expected-warning@+1 {{'__cdecl' calling convention is not supported for this target}}
+__inline __cdecl __attribute__((sycl_device)) int foo() { return 0; }
```

Comment:
Please, check that warning for `printf` is emitted if it's called from the SYCL kernel or device function.

Reply:
Yes, it still gets an error.